edit-client/src/client.rs

```diff
@@ -321,13 +322,18 @@ pub trait ClientImpl {
     where
         Self: Sized,
     {
-        self.send_client(&UserToFrontendCommand::Controls(Controls{
+        let mut controls_object = &UserToFrontendCommand::Controls(Controls{
             keys: key_handlers::<Self>()
                 .into_iter()
                 .map(|x| (x.0, x.1, x.2))
                 .collect(),
             buttons: button_handlers::<Self>(state).1
-        })).expect("Could not send initial state");
+        });
+
+        if Some(controls_object) != self.state().last_controls {
+            self.send_client(controls_object).expect("Could not send initial state");
```

Comment:
Here you can do `self.send_client(&UserToFrontendCommand::Controls(controls_object.clone()).expect(...)`.